tar/util.c

```diff
@@ -140,6 +140,7 @@ safe_fprintf(FILE *f, const char *fmt, ...)
 		} else {
 			/* Leave fmtbuff pointing to the truncated
 			 * string in fmtbuff_stack. */
+			fmtbuff = fmtbuff_stack;
```

Comment:
Ah.  No, you're right; if we had to grow the buffer more than once,`fmtbuff` would get left pointing to the previous heap buffer.

Good catch!